src/webui/www/private/views/cookies.html

```diff
@@ -0,0 +1,191 @@
+            </tr>
+        </thead>
+        <tbody>
+            <tr class="invisible newRow">
+                <td class="domain"><input type="text" aria-label="QBT_TR(Domain)QBT_TR[CONTEXT=CookiesDialog]" autocomplete="off" autocorrect="off" autocapitalize="none"></td>
+                <td class="path"><input type="text" aria-label="QBT_TR(Path)QBT_TR[CONTEXT=CookiesDialog]" autocomplete="off" autocorrect="off" autocapitalize="none"></td>
+                <td class="name"><input type="text" aria-label="QBT_TR(Name)QBT_TR[CONTEXT=CookiesDialog]" autocomplete="off" autocorrect="off" autocapitalize="none"></td>
+                <td class="value"><input type="text" aria-label="QBT_TR(Value)QBT_TR[CONTEXT=CookiesDialog]" autocomplete="off" autocorrect="off" autocapitalize="none"></td>
+                <td class="expDate"><input type="datetime-local" aria-label="QBT_TR(Expiration Date)QBT_TR[CONTEXT=CookiesDialog]"></td>
+                <td><img class="removeCookie" src="images/list-remove.svg" alt="QBT_TR(Remove)QBT_TR[CONTEXT=CookiesDialog]"></td>
+            </tr>
+        </tbody>
+        <tfoot>
+            <tr>
+                <td></td>
+                <td></td>
+                <td></td>
+                <td></td>
+                <td></td>
+                <td><img class="addCookie" src="images/list-add.svg" alt="QBT_TR(Add Cookie)QBT_TR[CONTEXT=CookiesDialog]"></td>
+            </tr>
+        </tfoot>
+    </table>
+
+    <div class="buttonContainer">
+        <button type="button" id="saveButton">QBT_TR(Save)QBT_TR[CONTEXT=HttpServer]</button>
+    </div>
+</div>
+
+<script>
+    "use strict";
+
+    window.qBittorrent ??= {};
+    window.qBittorrent.ManageCookies ??= (() => {
+        const exports = () => {
+            return {
+                setup: setup
+            };
+        };
+
+        const addCookie = function() {
+            const newRow = document.querySelector("#manageCookiesTable tr.newRow").cloneNode(true);
+            newRow.querySelector(".removeCookie").addEventListener("click", (event) => {
+                deleteCookie(event.target);
+            });
+            newRow.classList.remove("invisible");
+            document.querySelector("#manageCookiesTable tbody").append(newRow);
+            return newRow;
+        };
+
+        const deleteCookie = function(element) {
+            element.closest("tr").destroy();
+        };
+
+        const save = function() {
+            const rows = [...document.querySelectorAll("#manageCookiesTable tbody tr")].filter(e => !e.hasClass("invisible"));
+            const cookies = rows.map(row => {
+                const expDateValue = row.querySelector("td.expDate input").valueAsNumber;
+                // remove ms from string
+                const expDate = Number.isInteger(expDateValue) ? Number(String(expDateValue).slice(0, -3)) : 0;
+
+                return {
+                    domain: row.querySelector("td.domain input").value,
```

Comment:
I believe it has the same issue as this: https://github.com/qbittorrent/qBittorrent/pull/21321#discussion_r1759061804

Reply:
Wow, very surprising behavior. According to that article though, there's no performance boost in using `:scope`.

Reply:
>According to that article though, there's no performance boost in using :scope.

IMO it is more likely that browsers haven't optimized it yet.

I believe the 'no unexpected behavior' is more related here. There is no guarantee that the code won't select the wrong element in the future (when the html drastically changed).
If it isn't too much of a hassle, I would encourage you to add `:scope` here. Or at least, keep it in mind and use it in later PRs.

Reply:
In my testing I'm not able to reproduce this. Across Safari, Firefox, and Brave, calling `querySelector` and `querySelectorAll` on an element only queries that element's descendants. I've also tested that calling `document.getElementById("manageCookiesTable").querySelectorAll("body")` returns an empty list in all three browsers.

Safari
![Screenshot 2024-09-23 at 12 16 35](https://github.com/user-attachments/assets/d562f738-dcdd-4321-9646-e1e4c247f1dc)

Firefox
![Screenshot 2024-09-23 at 12 17 09](https://github.com/user-attachments/assets/4f58ff6e-d904-419f-a73e-589e408b5976)

Brave
![Screenshot 2024-09-23 at 12 17 47](https://github.com/user-attachments/assets/d1fe624a-3155-439d-9556-fe7743e3e0d4)

Reply:
>There is no guarantee that the code won't select the wrong element in the future (when the html drastically changed).

I was wrong about this. `element.querySelector` will only select element descendants.

The only difference with `:scope` is where does the matching start. With `:scope` it starts at the element. Without, it starts from the document root (and only considering the branch/path containing the element).
And since using `:scope` is slower (at least currently), IMO it should only be used when it is unavoidable.

Reply:
Still so strange that it doesn't always start at the element!